### PR TITLE
Promote pdb healthy to beta

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
The PodDisruptionBudget `spec.unhealthyPodEvictionPolicy` field has graduated to beta and is enabled by default. On servers with the feature enabled, this field may be set to `AlwaysAllow` to always allow unhealthy pods covered by the PodDisruptionBudget to be evicted.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3017-pod-healthy-policy-for-pdb
```

### CODE REVIEW

**Comment by cici37** [comment]
/remove-sig api-machinery
/sig node
Could you please add more details into PR description such like the KEP link, release notes, etc.? Thank you

**Comment by liggitt** [comment]
routing to @cici37 for initial review of the API update

**Comment by liggitt** [comment]
The KEP [beta graduation criteria](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3017-pod-healthy-policy-for-pdb#beta) mentions an e2e and conformance test. Can you update that to refer to an integration test (the functionality is fully contained within the API server, so is exercised by integration tests), and drop the reference to a conformance test (we don't conformance test beta features).

/lgtm
/approve

File: pkg/apis/policy/types.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ type PodDisruptionBudgetSpec struct {
 	// Clients making eviction decisions should disallow eviction of unhealthy pods
 	// if they encounter an unrecognized policy in this field.
 	//
-	// This field is alpha-level. The eviction API uses this field when
+	// This field is beta-level. The eviction API uses this field when
 	// the feature gate PDBUnhealthyPodEvictionPolicy is enabled (disabled by default).
```
Comment (atiratree) [inline pkg/apis/policy/types.go:70]:
```suggestion
    // the feature gate PDBUnhealthyPodEvictionPolicy is enabled (enabled by default).
```

@ravisantoshgudimetla these need to be updated as well

**Comment by atiratree** [comment]
> The KEP [beta graduation criteria](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3017-pod-healthy-policy-for-pdb#beta) mentions an e2e and conformance test. Can you update that to refer to an integration test (the functionality is fully contained within the API server, so is exercised by integration tests), and drop the reference to a conformance test (we don't conformance test beta features).

@liggitt sorry my confusing wrt this, does this mean we do not need to introduce e2e tests for GA since it is an only API server feature?

**Comment by ravisantoshgudimetla** [comment]
> The KEP [beta graduation criteria](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3017-pod-healthy-policy-for-pdb#beta) mentions an e2e and conformance test. Can you update that to refer to an integration test (the functionality is fully contained within the API server, so is exercised by integration tests), and drop the reference to a conformance test (we don't conformance test beta features).
> 
> /lgtm /approve

Opened https://github.com/kubernetes/enhancements/pull/3883

File: pkg/apis/policy/types.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/apis/policy/types.go:70]: Thank you for catching it @atiratree. Opened https://github.com/kubernetes/kubernetes/pull/115914 to address the same.

**Pull request merged**